File: 0515-find-largest-value-in-each-tree-row/0515-find-largest-value-in-each-tree-row.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> largestValues(TreeNode root) {
        if (root == null) {
            return new ArrayList<Integer>();
        }
        Queue<TreeNode> q = new LinkedList<>();
        q.add(root);
        List<Integer> ret = new ArrayList<>();
        while(!q.isEmpty()) {
            final int size = q.size();
            List<Integer> nums = new ArrayList<>();
            
            for (int i=0; i<size; i++) {
                TreeNode current = q.poll();
                if (current != null) {
                    nums.add(current.val);
                    if (current.left != null)   q.add(current.left);
                    if (current.right != null)   q.add(current.right);
                }
            }

            ret.add(getMax(nums));
        }
        

        return ret;
    }

    public int getMax(List<Integer> list) {
        int max = Integer.MIN_VALUE;

        for (int x : list) {
            max = x > max ? x : max;
        }
        return max;
    }
}

File: 0515-find-largest-value-in-each-tree-row/0515-find-largest-value-in-each-tree-row.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var largestValues = function(root) {
    if (root == null) return []
    let ret = []
    const q = []
    q.push(root)

    while(q.length != 0) {
        const size = q.length
        const level = []

        for (let i = 0; i < size; i++) {
            const curr = q.shift()
            if (curr) {
                if (curr.left) {q.push(curr.left)}
                if (curr.right) {q.push(curr.right)}
                level.push(curr.val)
            }
        }
        ret.push(Math.max(...level))
    }

    return ret
};